src/main/java/org/commcare/xml/CaseXmlParser.java

```diff
@@ -187,6 +187,10 @@ private void updateCase(Case caseForBlock, String caseId) throws InvalidStructur
                 case "owner_id":
                     String oldUserId = caseForBlock.getUserId();
 
+                    if (oldUserId == null) {
+                        throw InvalidStructureException.readableInvalidStructureException("The case_id field of a <case> was found null", parser);
```

Comment:
In the error message `userid` should be null instead of case_id ? 

Also in the jira issue, seems like `owner_id` was not defined while from the error stack it seems like `userid` is getting set to null. So I am a bit confused what exactly is happening.

Reply:
Yeah, I'm also not sure how owner_id is different from userid. https://github.com/dimagi/commcare-core/blob/master/src/main/java/org/commcare/cases/model/Case.java#L209

Reply:
Yeah [seems like](https://github.com/dimagi/commcare-core/blob/master/src/main/java/org/commcare/xml/CaseXmlParser.java#L167) they are getting used interchangeably on phone with the assumption that case creator is always the owner of the case.  I think we should be doing this `null` check at the [case create](https://github.com/dimagi/commcare-core/blob/master/src/main/java/org/commcare/xml/CaseXmlParser.java#L167) part itself and throw the error from there.

Reply:
Cuz of [this condition](https://github.com/dimagi/commcare-core/blob/master/src/main/java/org/commcare/xml/CaseXmlParser.java#L146) It seems like `owner_id` is left out deliberately?

Reply:
Hmm, I am not really sure. Lets ask @ctsims